desc/compute/_curve.py

```diff
@@ -653,34 +654,83 @@ def _x_sss_FourierXYZCurve(params, transforms, profiles, data, **kwargs):
     basis="{'rpz', 'xyz'}: Basis for returned vectors, Default 'rpz'",
 )
 def _x_SplineXYZCurve(params, transforms, profiles, data, **kwargs):
+
     xq = data["s"]
+    transforms["intervals"] = jnp.asarray(transforms["intervals"])
 
-    Xq = interp1d(
-        xq,
-        params["knots"],
-        params["X"],
-        method=transforms["method"],
-        derivative=0,
-        period=2 * jnp.pi,
-    )
-    Yq = interp1d(
-        xq,
-        params["knots"],
-        params["Y"],
-        method=transforms["method"],
-        derivative=0,
-        period=2 * jnp.pi,
-    )
-    Zq = interp1d(
-        xq,
-        params["knots"],
-        params["Z"],
-        method=transforms["method"],
-        derivative=0,
-        period=2 * jnp.pi,
-    )
+    is_discontinuous = len(transforms["intervals"][0])
+
+    def get_f_in_interval(arr, knots, istart, istop):
+        arr_temp = jnp.where(knots > knots[istop], arr[istop], arr)
+        arr_temp = jnp.where(knots < knots[istart], arr[istart], arr_temp)
+
+        return arr_temp
+
+    def inner_body(x, y, z, knots, period=None):
```

Comment:
to reduce duplicate code, you could define this once, outside of any single compute function, with an additional kwarg for `derivative`. Then you can just call `inner_body` for each of the compute fxns with the correct `derivative` order

Reply:
you can also do the same for `get_f_in_interval` and add `derivative` as an argument to that too then you should only need one function call of it per compute function, which should redue a lot of thestuff